domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -190,14 +180,41 @@ private static SecurityDomain toSecurityDomain(final FutureTask<SecurityDomain>
         }
     }
 
+    private static FutureTask<SecurityDomain> toFutureTask(InjectedValue<SecurityDomain> injectedValue) {
+        return new FutureTask<>((Callable<SecurityDomain>)  (() -> injectedValue.getValue()));
+    }
+
     static class IdentityService implements Service<Void> {
 
         private List<FutureTask<SecurityDomain>> getSecurityDomainFutures = Collections.emptyList();
```

Comment:
This should just be List<InjectedValue<SecurityDomain>>.

These FutureTasks are now just a really complicated way of calling injectedValue.getValue(). The IdentityService depends on all the security domains so when IdentityService.start() is called there is no "future" work involved any more.